### PR TITLE
add ai qa bot

### PR DESCRIPTION
**Pull request by nobodyiam**
## What's the purpose of this PR

Integrate an intelligent QA bot capable of providing natural language responses to users' inquiries.
The source code for the backend Apollo QA bot can be accessed at [here](https://github.com/apolloconfig/apollo-qa-bot) and is currently deployed in conjunction with the Apollo demo server, which is why the ip is 81.68.181.139.

### Screenshots

image

image

image

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/apolloconfig/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [ ] Update the [`CHANGES` log](https://github.com/apolloconfig/apollo/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by Anilople** [comment]
work well.

image

File: docs/index.html [hunk 1]
```diff
@@ -151,6 +156,10 @@
         generateMultipleLanguagesNavbarPluginByListItemName("Translations"),
       ],
     };
+    QABot.initialize({
+      "serverUrl": "http://81.68.181.139/ai/qa",
```
Comment (Anilople) [inline docs/index.html:160]: For simulating the ai backend shutdown, 

I change 

`http://81.68.181.139` to `http://81.68.181.100`

The document can be open, but would delay 10s . 

Is there any way to setup the timeout from 10s to 5s, 3s or 2s?

**Review by Anilople — DISMISSED** [review]

**Comment by Anilople** [comment]
Maybe we can change the logo

image

in the feature to attract user to click it more frequently.

* Add some text to it
* Zoom out it to get larger logo

File: docs/index.html (hunk 1 above)
Comment (nobodyiam) [reply docs/index.html]: This is a good point! I changed the logic as follows and it should work well when 81.68.181.139 is down.

```javascript
<!-- add qa bot -->
<script>
  (function() {
    var link = document.createElement('link');
    link.rel = 'stylesheet';
    link.href = 'http://81.68.181.139/ai/qa-bot.css';
    document.head.appendChild(link);

    var script = document.createElement('script');
    script.src = 'http://81.68.181.139/ai/qa-bot.js';
    script.onload = function () {
      QABot.initialize({
        "serverUrl": "http://81.68.181.139/ai/qa",
        "documentSiteUrlPrefix": "https://www.apolloconfig.com/#"
      });
    };
    document.body.appendChild(script);
  })();
</script>
```

**Comment by nobodyiam** [comment]
I added the `Chat with AI` text below the icon.
image

File: docs/index.html (hunk 1 above)
Comment (Anilople) [reply docs/index.html:160]: Good Idea.

But it is still block on my machine...

Found that
```
  <link rel="stylesheet" href="http://81.68.181.139/ai/qa-bot.css">
  <script src="http://81.68.181.139/ai/qa-bot.js"></script>
```
i.e those 2 lines cause.

When change them to

```
  <link rel="stylesheet" href="http://81.68.181.100/ai/qa-bot.css">
  <script src="http://81.68.181.100/ai/qa-bot.js"></script>
```

We can reproduce it.

----

After query how to async load css and javascript through ChatGPT, There is a solution

use 
```
  <link rel="preload" href="http://81.68.181.139/ai/qa-bot.css" as="style">
  <noscript><link rel="stylesheet" href="http://81.68.181.139/ai/qa-bot.css"></noscript>
```
to async load css, and use

```
  <script src="http://81.68.181.139/ai/qa-bot.js" async></script>
```
to async load javascript.

Comment (nobodyiam) [reply docs/index.html]: I'm sorry, looks like the last push failed, these 2 lines should be removed.

<link rel="stylesheet" href="http://81.68.181.139/ai/qa-bot.css">
<script src="http://81.68.181.139/ai/qa-bot.js"></script>

**Review by Anilople — APPROVED** [review]
LGTM

**Pull request merged**